src/Components/PostsWrapper/PostsWrapper.jsx

```diff
@@ -0,0 +1,189 @@
+    display,
+    gridTemplateColumns,
+    columnGap,
+    rowGap,
+    textAlign,
+    fontSize,
+    color,
+    height,
+    width,
+    margin,
+    padding,
+    position,
+    objectFit,
+    backgroundColor,
+    fontFamily,
+    overflow,
+    bottom
+}) => {
+    const post1 = {
+        id: 1,
+        title:
+      'Lorem Ipsum is simply dummy ',
+        imageURL:
+      'https://images.unsplash.com/photo-1653959023171-746709105833?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=735&q=80',
+        date: '2022-05-08'
+    }
+    const post2 = {
+        id: 2,
+        title: 'Lorem Ipsum is simply dummy text of the printing and typesetting',
+        imageURL:
+      'https://images.unsplash.com/photo-1471897488648-5eae4ac6686b?ixlib=rb-1.2.1&ixid=MnwxMjA3fDF8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=687&q=80',
+        date: '2022-05-12'
+    }
+    const post3 = {
+        id: 3,
+        title: 'Lorem Ipsum is simply dummy text of the printing',
+        imageURL:
+      'https://images.unsplash.com/photo-1602447668419-c22aa459221c?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=778&q=80',
+
+        date: '2022-05-20'
+    }
+    const post4 = {
+        id: 4,
+        title: 'Lorem Ipsum is simply dummy text of the printing',
+        imageURL:
+      'https://images.unsplash.com/photo-1653908660985-338550642ac2?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1609&q=80',
+        date: '2022-05-28'
+    }
+    const post5 = {
+        id: 5,
+        title: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit',
+        imageURL:
+      'https://images.unsplash.com/photo-1602447668419-c22aa459221c?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=778&q=80',
+        date: '2022-06-01'
+    }
+    const post6 = {
+        id: 6,
+        title: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit',
+        imageURL:
+      'https://images.unsplash.com/photo-1602447668419-c22aa459221c?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=778&q=80',
+        date: '2022-06-01'
+    }
+    const post7 = {
+        id: 7,
+        title: 'Lorem Ipsum is simply dummy text of the printing',
+        imageURL:
+      'https://images.unsplash.com/photo-1653908660985-338550642ac2?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1609&q=80',
+        date: '2022-05-28'
+    }
+
+    const postai = [post1]
+    // const postai = [post1, post2]
+    // const postai = [post1, post2, post3]
+    // const postai = [post1, post2, post3, post4]
+    // const postai = [post1, post2, post3, post4, post5, post6, post7]
+
+    switch (postai.length) {
+    case 0:
+        return (
+        // TODO:
+            <p>No Posts added</p>
+        )
+    case 1:
+        return (
+            <OnePost
```

Comment:
I don't know if I am right, but your code is very repetitive. Maybe it is possible to create only one component to which you can pass all these props and make calculations there?

Reply:
I assume, such component would become quite complicated and not easy to understand. Also the code will be much longer. Arunas left us possibility to choose whether it will be one big component or several smaller ones. We chose second approach :)